### PR TITLE
GEM-85 | Add get for exchange rate

### PR DESCRIPTION
**Pull request by lawos98**

### CODE REVIEW

File: src/test/kotlin/pl/edu/agh/gem/external/dto/InternalExchangeRateResponseTest.kt [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+import io.kotest.core.spec.style.ShouldSpec
+import io.kotest.matchers.shouldBe
+import pl.edu.agh.gem.internal.model.ExchangeRate
+import java.math.BigDecimal
+import java.time.Instant.parse
+
+class InternalExchangeRateResponseTest : ShouldSpec({
+
+    should("correctly map ExchangeRate to ExternalExchangeRateResponse") {
```
Comment (pwgacek) [inline src/test/kotlin/pl/edu/agh/gem/external/dto/InternalExchangeRateResponseTest.kt]:
```suggestion
    should("correctly map ExchangeRate to InternalExchangeRateResponse") {
```

File: src/main/kotlin/pl/edu/agh/gem/external/dto/ExternalExchangeRateResponse.kt [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+import pl.edu.agh.gem.internal.model.ExchangeRate
+import java.math.BigDecimal
+import java.time.Instant
+
+data class ExternalExchangeRateResponse(
+    val currencyFrom: String,
+    val currencyTo: String,
+    val rate: BigDecimal,
+    val createAt: Instant,
```
Comment (pwgacek) [inline src/main/kotlin/pl/edu/agh/gem/external/dto/ExternalExchangeRateResponse.kt]:
```suggestion
    val createdAt: Instant,
```

File: src/main/kotlin/pl/edu/agh/gem/external/dto/InternalExchangeRateResponse.kt [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+import pl.edu.agh.gem.internal.model.ExchangeRate
+import java.math.BigDecimal
+import java.time.Instant
+
+data class InternalExchangeRateResponse(
+    val currencyFrom: String,
+    val currencyTo: String,
+    val rate: BigDecimal,
+    val createAt: Instant,
```
Comment (pwgacek) [inline src/main/kotlin/pl/edu/agh/gem/external/dto/InternalExchangeRateResponse.kt]:
```suggestion
    val createdAt: Instant,
```

File: src/main/kotlin/pl/edu/agh/gem/internal/model/ExchangeRate.kt [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package pl.edu.agh.gem.internal.model
+
+import java.math.BigDecimal
+import java.time.Instant
+
+data class ExchangeRate(
+    val currencyFrom: String,
+    val currencyTo: String,
+    val rate: BigDecimal,
+    val createAt: Instant,
```
Comment (pwgacek) [inline src/main/kotlin/pl/edu/agh/gem/internal/model/ExchangeRate.kt]:
```suggestion
    val createdAt: Instant,
```

File: src/test/kotlin/pl/edu/agh/gem/util/TestHelper.kt [hunk 5]
```diff
@@ -1 +1,17 @@
...
+
+import pl.edu.agh.gem.internal.model.ExchangeRate
+import java.time.Instant
+import java.time.Instant.parse
+
+fun createExchangeRate(
+    currencyFrom: String = "USD",
+    currencyTo: String = "EUR",
+    rate: String = "1.0",
+    createAt: Instant = parse("2023-01-01T00:00:00.00Z"),
```
Comment (pwgacek) [inline src/test/kotlin/pl/edu/agh/gem/util/TestHelper.kt]:
```suggestion
    createdAt: Instant = parse("2023-01-01T00:00:00.00Z"),
```

**Review by pwgacek — APPROVED** [review]

File: src/main/kotlin/pl/edu/agh/gem/external/dto/InternalExchangeRateResponse.kt [hunk 6]
```diff
@@ -0,0 +1,19 @@
+package pl.edu.agh.gem.external.dto
+
+import pl.edu.agh.gem.internal.model.ExchangeRate
+import java.math.BigDecimal
+import java.time.Instant
+
+data class InternalExchangeRateResponse(
```
Comment (pwgacek) [inline src/main/kotlin/pl/edu/agh/gem/external/dto/InternalExchangeRateResponse.kt:7]: do we want to return currencyFrom, currencyTo, createdAd?  In expense manager i only need value of rate field

Comment (pwgacek) [inline src/main/kotlin/pl/edu/agh/gem/external/dto/InternalExchangeRateResponse.kt:7]: do we want to return currencyFrom, currencyTo, createdAd? In expense manager i only need value of rate field

**Review by pwgacek — CHANGES_REQUESTED** [review]

File: src/main/kotlin/pl/edu/agh/gem/external/dto/InternalExchangeRateResponse.kt (hunk 6 above)
Comment (lawos98) [reply src/main/kotlin/pl/edu/agh/gem/external/dto/InternalExchangeRateResponse.kt:7]: only for logs and eventualy debuging

**Review by pwgacek — APPROVED** [review]

**Pull request merged**